src/engine.c

```diff
@@ -278,6 +278,11 @@ int init_context(struct app_config *app_config, struct supervisor_context *ctx)
   }
 
   utarray_new(ctx->config_ifinfo_array, &config_ifinfo_icd);
+  if (app_config->config_ifinfo_array == NULL) {
```

Comment:
Potentially a dumb `C` question, but can `app_config->config_ifinfo_array` be uninitialized, and therefore be filled with random data??

Not sure how to check that it's not a valid `utarray` without a segmentation fault, though.

Reply:
`app_config->config_ifinfo_array` can be freed and then it is counted as uninitialized.

`utarray` library does check. On allocation error it exist the process https://github.com/nqminds/EDGESec/blob/main/src/utils/utarray.h#L108

Reply:
Correct me if I'm wrong, (last time I programmed in C was ages ago), but if I do the following:

### `app_config.config_ifinfo_array` is unitialized (junk data)

Maybe undefined behaviour?? I'm not sure if C can catch this issue.

```c
app_config app_config;
app_config.param1 = 1;
init_context(&app_config, ...)
```

### `app_config.config_ifinfo_array` is initalized to 0 (NULL ptr)

Causes a segmentation fault at the following line (which is why I added the `if()` check)

https://github.com/nqminds/EDGESec/blob/99bedafa65c5519207408b79504df4cf1f59d3fa/src/engine.c#L281

```c
app_config app_config = {0, .param1 = 1};
init_context(&app_config, ...)
```

### `app_config.config_ifinfo_array` is correctly initalized to a utarray

Expected valid behaviour by `init_context`

```c
app_config app_config = {0, .param1 = 1};
utarray_new(app_config.config_ifinfo_array, &ut_str_icd);
init_context(&app_config, ...)
```

## Potential cleanup

Instead of having a heap-allocated utarray using `UT_array*` and using `utarray_new` or `utarray_free`, maybe it's better to have a stack-allocated `UT_array` (e.g. not a pointer), and using `utarray_init` and `utarray_done` instead? 

https://github.com/nqminds/EDGESec/blob/56299399c6f8d8eed97397ea675e496160189ec6/src/config.h#L47